splash/qtutils.py

```diff
@@ -166,7 +166,7 @@ def qt2py(obj, max_depth=100):
 
     # print(obj, obj.__class__)
 
-    if isinstance(obj, QString):
+    if isinstance(obj, (QString,QByteArray)):
         return unicode(obj)
```

Comment:
Hmmm... if I decorate `resume()` with a `QString` argument, then `null` values in JavaScript will turn into empty strings `''` in Python. I decorated with `QByteArray` because it does not interfere with `null`. However I tried testing with non-ASCII characters this morning and QT converts from native JavaScript string encoding (UTF-16) to ASCII before it calls our slot method! So if JavaScript returns '你好', then the Python slot method sees '??' as its `value` argument. Argh...

One possible simplification (at the expense of making the API more complex) is to have multiple resume methods, each of which handles a specific type, e.g. `resume_number()`, `resume_array()`, etc. Or alternatively, we can force the caller to pass a specific data type, e.g. `resume()` must take a JavaScript object parameter. I know its not ideal, but QT's metatype system is so screwy that I think

In the meantime, I'll keep investigating how to correctly pass strings from JavaScript back to Python.